src/main/java/ratpack/zipkin/TracedParallelBatch.java

```diff
@@ -60,8 +60,7 @@ private TracedParallelBatch(Iterable<? extends Promise<T>> promises) {
    */
   public static <T> ParallelBatch<T> of(final TraceContext context, Iterable<? extends Promise<T>> promises) {
     return ParallelBatch.of(promises)
-        .execInit(execution ->
-            execution.add(new RatpackCurrentTraceContext.TraceContextHolder(context)));
+        .execInit(execution -> execution.add(RatpackCurrentTraceContext.wrap(context)));
```

Comment:
Really wish this wasn't referring to `RatpackCurrentTraceContext` directly here.  It basically makes it impossible to use a different implementation.

Long term I think this API probably needs some deeper thought.  For now I'm just punting on it but should probably make a ticket and try to come up with a better solution.

Reply:
@llinder by "different implementation", do you mean a different implementation of `CurrentTraceContext`?

Reply:
Yeah, meant different implementation of CurrentTraceContext.  Right now this is tightly coupled to RatpackCurrentTraceContext which in hind sight feels less than ideal. :(

Reply:
@llinder what use case can you think of where we'd require a different implementation of `CurrentTraceConext`? In any case, as long as this stuff is an implementation detail of (in this case `TraceParallelBatch` and it doesn't leak out, I figure we're free to change it later if we feel it's necessary.

Reply:
An example would have been if I needed a custom CurrentTraceContext that does some magic for non Ratpack things.  One case is the MDC addition but there are others such as some cross execution stuff between Ratpack and Cassandra.  In order to solve the Ratpack Execution -> Cassandra Execution hops I'm likely going to have to modify the CurrentTraceContext.  Honestly not sure how yet but I know its been a problem for us so I will be looking into it more soon.

In general though, the CurrentTraceContext is pluggable in Brave, so maintaining the "pluggability" in the Ratpack instrumentation would be ideal.